cedar-policy-core/src/tpe/response.rs

```diff
@@ -92,10 +97,14 @@ pub struct Response<'a> {
     false_forbids: HashSet<PolicyID>,
     // All of the [`Effect::Forbid`] policies that evaluated to a residual
     non_trivial_forbids: HashSet<PolicyID>,
-    // request
-    request: &'a PartialRequest,
-    // entities
-    entities: &'a PartialEntities,
+    // TODO we could make this owned, but would require a clone
+    // request used for this partial evaluation
+    // this is [`None`] for batched evaluation
+    request: Option<&'a PartialRequest>,
+    // TODO we could make this owned, but would require a clone
+    // entities used for this partial evaluation, if any
+    // this is [`None`] for batched evaluation
+    entities: Option<&'a PartialEntities>,
```

Comment:
This is awkward, but I didn't want to introduce a clone here. It's used for sanity checking.